catalog/imaging/imaging-transformer-chipping/src/main/java/org/codice/alliance/imaging/chip/transformer/CatalogOutputAdapter.java

```diff
@@ -179,7 +179,9 @@ public BinaryContent getBinaryContent(BufferedImage image)
   public NitfSegmentsFlow getNitfSegmentsFlow(ResourceResponse resourceResponse)
       throws NitfFormatException, IOException {
     notNull(resourceResponse, "resourceResponse must be non-null");
-    return getNitfSegmentsFlow(resourceResponse.getResource().getInputStream());
+    try (final InputStream inputStream = resourceResponse.getResource().getInputStream()) {
```

Comment:
We shouldn't be using try-with-resources on input stream. If an error occurs when closing the stream, that error will bubble out. However, in such situations, there is nothing that can be done anyway and the stream has been read as designed. It is better to handle the stream in a finally close with something like IOUtils.closeQuiertly().
Output streams should always be done via try-with-resources.

Reply:
May be out of scope, but is there any concern with `resourceResponse.getResource()` being null?

Reply:
Probably should add a second notnull check.